File: 1971-find-if-path-exists-in-graph/1971-find-if-path-exists-in-graph.cpp
class Solution {
public:
    bool validPath(int n, vector<vector<int>>& edges, int source, int destination) {
        unordered_map<int,vector<int>>G;
        for(auto e:edges){
            G[e[0]].push_back(e[1]);
            G[e[1]].push_back(e[0]);
        }
        queue<int>q;
        vector<bool>visited(n,false);
        q.push(source);
        
        while(!q.empty()){
            int x=q.front();
            q.pop();
            if(x==destination)
                return true;
            visited[x]=true;
            for(int i:G[x]){
                if(!visited[i]){
                    visited[i]=true;
                    q.push(i);
                }
            }
        }
        return false;
    }
};

File: 1971-find-if-path-exists-in-graph/1971-find-if-path-exists-in-graph.py
class Solution:
    def validPath(self, n: int, edges: List[List[int]], source: int, destination: int) -> bool:
        adj=[[] for i in range(n)]
        for i,j in edges:
            adj[i].append(j)
            adj[j].append(i)
        status=[0]*n
        q=[]
        q.append(source)
        status[source]=1
        while len(q)>0:
            x=q.pop(0)
            l=adj[x]
            for y in l:
                if status[y]==0:
                    status[y]=1
                    q.append(y)

        if status[destination]==0:
            return False
        return True

        